helm: disable PSPs on 1.23 and document PSA usage (#21258)
gravitational/teleport: docs/pages/deploy-a-cluster/helm-deployments/migration-kubernetes-1-25-psp.mdx

---
title: Kubernetes 1.25 and PSP removal

> description: How to prepare the PodSecurityPolicy removal happening in Kubernetes 1.25

Reviewer: ```suggestion
description: How to prepare for the PodSecurityPolicy removal happening in Kubernetes 1.25
```

---

PodSecurityPolicies (PSPs) were deprecated in Kubernetes 1.22 and are removed
in Kubernetes 1.25. This page explains the security implications of such changes
for Teleport users, and what actions are required.

The two Teleport charts `teleport-cluster` and `teleport-kube-agent` were relying
on PodSecurityPolicies to provide an additional security level for Teleport pods.
Their removal has two main consequences:

- After upgrading to 1.25, Helm can end up with a corrupted state referencing

>   PSP objects. In this case, state has to be manually fixed.

Reviewer 2: is `state` an ephemeral concept, or should this be something like:

```suggestion
  PSP objects. In this case, the cluster state has to be manually fixed.
```

Author: ```suggestion
  PSP objects. In this case, the Helm release state has to be manually fixed.
```

> - Security policy enforcement is managed by PodSecurityAdmission (PSA) since 1.23.they are
>   PSA security level is configured on the `namespace` resource, which Helm don't manage.
>   You now have to set security enforcement level as the chart can't do it.

Reviewer 2: I think, but I'm not sure:

```suggestion
- Security policy enforcement is managed by PodSecurityAdmission (PSA) since 1.23.
  PSA security level is configured on the `namespace` resource, which Helm doesn't manage.
  You now have to set the security enforcement level as the chart can't do it.
```
